docs/howto_contribute.md

```diff
@@ -0,0 +1,396 @@
+
+- name: Copy rhel_makecache.sh to temp directory
+  copy:
+    src: rhel_makecache.sh
+    dest: "{{ build_dir }}"
+    owner: root
+    group: root
+    mode: 0744
+  when: ansible_distribution == 'RedHat'
+
+- name: Build httpd image
+  command: "docker build -t {{ g_httpd_name }} -f {{ build_dir }}/{{ g_httpd_name }}_Dockerfile {{ build_dir }}"
+
+- name: Get docker images after build
+  command: docker images
+  register: build_images
+
+- name: Fail if httpd image not present
+  fail:
+    msg: "The {{ g_httpd_name }} image is not present"
+  when: "'{{ g_httpd_name }}' not in build_images.stdout"
+```
+
+In this example, the role makes an assumption about the name of the Dockerfile,
+the existence of a `makecache.sh` script, and relies on a global variable
+throughout.
+
+This role could be improved by removing the assumption about the `makecache.sh`
+script, parameterizing more of the tasks, and separating out tasks that
+manipulate files/directories into separate roles/tasks.  The improved version
+should focus on just the `docker build` command.
+
+An improved version would live in `roles/docker_build/tasks/main.yml` and
+could look like this:
+
+```yaml
+---
+- name: Fail if required variables are not set
+  fail:
+    msg: "The required variables are not set."
+  when: image_name is not defined or
+        build_path is not defined or
+        dockerfile_name is not defined
+
+- name: Build Docker image
+  command: "docker build -t {{ image_name }} -f {{ build_path }}/{{ dockerfile_name }} {{ build_path }}"
+
+- name: Verify Docker image was built
+  command: docker images
+  register: docker_images
+  failed_when: "'{{ image_name }}' not in docker_images.stdout"
+```
+
+This completely simplifies the role to handle the building of a Docker image
+when given the required parameters.  In this simplified version, you would
+have to have separate roles/tasks that would handle the creation of any
+required directories and copying the Dockerfile to the host.
+
+### Writing Your First Test
+
+Now you have a new role created, let's put it to use in a new test playbook.
+
+#### Directory Setup
+
+Firstly, you will have to create a new sub-directory under `tests` and create
+symlinks to the top-level `callback_plugins` and `roles` directories.  This
+allows the test to reference the available roles and have its output nicely
+formatted by the callback plugin.  In this example, we are going to pretend we
+are creating a test playbook for the `docker build` command.
+
+```bash
+$ mkdir -p tests/docker-build/
+$ ln -s ../../callback_plugins/ tests/docker-build/callback_plugins
+$ ln -s ../../roles/ tests/docker-build/roles
+```
+
+Additional directories such as `files`, `templates` or `vars` may be required
+for your test.  Please reference the [Ansible documentation](http://docs.ansible.com/ansible/playbooks_best_practices.html#content-organization) about
+content organization about when and how to use these directories.
+
```

Comment:
Maybe stick a quick blurb in here about setting up ``redhat_subscription/files/subscription_data.csv.sample``?

Which reminds me, that may be a good thing to add to your ``.gitignore`` file, ``subscription_data.csv``, good to prevent obvious woopsies where you can.